src/main/java/reactor/netty/channel/MonoSendMany.java

```diff
@@ -140,6 +141,7 @@ public Object scanUnsafe(Attr key) {
 		SendManyInner(MonoSendMany<I, O> parent, CoreSubscriber<? super Void> actual) {
 			this.parent = parent;
 			this.actual = actual;
+			this.actualContext = actual.currentContext();
```

Comment:
so now we have a field capturing the `Context` of downstream... I'm confused now, as you opposed doing that in the prveious PR when I suggested we store the context as  `this.context = actual.currentContext().put(KEY_ON_DISCARD, this)` in the constructor ?

Reply:
yeah. it is confusing. The difference is that

`this.actualContext = actual.currentContext();`

is obtaining already created object whilst

`this.actualContext = actual.currentContext().put(...);`

create a new one along with the previous.


The idea here is not to produce a new object but store `actual.currentContext();` since underneath  it calls 

https://github.com/reactor/reactor-netty/blob/master/src/main/java/reactor/netty/channel/ChannelOperations.java#L503

which in turn end up calling 

https://github.com/reactor/reactor-netty/blob/41e73261a5935dac3b5773ab3668d0dd99d4cf42/src/main/java/reactor/netty/resources/DefaultPooledConnectionProvider.java#L364.

thus, in order to avoid that invocation all the time, we resolve context once at the construction

Reply:
note that `owner()` checks always a channel attribute which is an expensive operation
https://github.com/reactor/reactor-netty/blob/41e73261a5935dac3b5773ab3668d0dd99d4cf42/src/main/java/reactor/netty/resources/DefaultPooledConnectionProvider.java#L445-L462